### PR TITLE
fix: move/delete/append actions on tree 

### PR DESCRIPTION
**Pull request by SamuelAlev**

### CODE REVIEW

File: src/components/Tree/types.ts [hunk 1]
```diff
@@ -90,7 +90,7 @@ export type TreeState = {
     expandedIds: Set<string>;
     selectionMode: 'single' | 'multiselect';
     overlay?: Overlay;
-    nodes: ReactElement[];
+    nodes: ReactElement<InternalTreeItemProps>[];
```
Comment (SamuelAlev) [inline src/components/Tree/types.ts:93]: Let's keep it like that and fix the type issues by checking the data :)

File: src/components/Tree/Tree.tsx [hunk 2]
```diff
@@ -129,16 +139,13 @@ const reducer = produce((draft: TreeState, action: TreeStateAction) => {
                     return;
                 }
 
+                const offset = draft.nodes.findLastIndex((node) => node.props.parentId === id) - sliceIndex;
```
Comment (anxobotana) [inline src/components/Tree/Tree.tsx:142]: Note that when the item is deleted `node.props.parentId` still contains the old data, so I think this won't exclude it from the list.
That's why I added the filter part:
```
((node.props.parentId === id && newNodeChildrenIds.includes(node.props.id)) || node.props.parentId !== id)
```

Comment (anxobotana) [reply src/components/Tree/Tree.tsx:142]: nvm... I see what you did... it might work 😄

File: src/components/Tree/Tree.tsx [hunk 3]
```diff
@@ -122,28 +121,24 @@ const reducer = produce((draft: TreeState, action: TreeStateAction) => {
...
+                if (parentNodeIndex === -1) {
+                    console.error(`Element with ID "${parentId}" not found.`);
                     return;
                 }
 
-                const currentChildrenIds = getNodeChildrenIds(draft.nodes, id);
+                const currentChildrenIds = getNodeChildrenIds(draft.nodes, parentId);
 
-                const newChildrenIds = children.map((node) => node.props.id as string);
+                const newChildrenIds = children.map((node) => node.props.id);
```
Comment (GoranRibic) [inline src/components/Tree/Tree.tsx:135]: Fyi this will have type any[] if we remove 'as string'.

**Review by GoranRibic — APPROVED** [review]

File: src/components/Tree/Tree.tsx (hunk 3 above)
Comment (SamuelAlev) [reply src/components/Tree/Tree.tsx:135]: Not anymore
Screenshot 2023-05-08 at 09 19 47

**Pull request merged**